src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -113,12 +115,13 @@ private Task<Document> UpdateDocumentAsync(Document document, SyntaxNode root, S
             var previousContentWasStringLiteralExpression = false;
             foreach (var piece in pieces)
             {
-                var currentContentIsStringLiteral = syntaxFacts.IsStringLiteralExpression(piece);
-                if (currentContentIsStringLiteral)
+                var isCharacterLiteral = syntaxFacts.IsCharacterLiteralExpression(piece);
+                var currentContentIsStringOrCharacterLiteral = syntaxFacts.IsStringLiteralExpression(piece) || isCharacterLiteral;
+                if (currentContentIsStringOrCharacterLiteral)
                 {
                     var text = piece.GetFirstToken().Text;
                     var textWithEscapedBraces = text.Replace("{", "{{").Replace("}", "}}");
-                    var textWithoutQuotes = GetTextWithoutQuotes(textWithEscapedBraces, isVerbatimStringLiteral);
+                    var textWithoutQuotes = GetTextWithoutQuotes(textWithEscapedBraces, isVerbatimStringLiteral, isCharacterLiteral);
```

Comment:
dumb question. how does this work if you have a normal char lit with something like `\u000a` and you have that in either an `$""` interpolated string or an `@$""` interpolated string?  Presumably we need to do some escaping for one or hte other.

Reply:
We don't interpolate verbatim strings with non verbatim

Reply:
That is the case with strings.  But you're adding support for char literals (which have no verbatim/non-verbatim split).  How are you ensuring the char literal form matches the string form?

Reply:
Line 64 we add char literals to the list of string literals. Later on we check that if any are verbatim, all are verbatim.

Reply:
So you simply cannot concatenate a char with a verbatim string. I think this is fair enough, as the same would be true for a one letter non verbatim string, whatever it's contents.

Reply:
got it.  do we have a test for this?  Thanks!

Reply:
Yes TestMissingWithMixedStringTypes3 in ConvertConcatenatedToInterpolatedStringTests.cs

Reply:
👍